yt_dlp/extractor/cda.py

```diff
@@ -101,7 +101,9 @@ def _download_age_confirm_page(self, url, video_id, *args, **kwargs):
             }, **kwargs)
 
     def _perform_login(self, username, password):
-        cached_bearer = self.cache.load(self._BEARER_CACHE, username) or {}
+        # hack to allow e-mails as a key
+        cache_username = username.replace('@', '___')
```

Comment:
better to modify `Cache._get_cache_fn` instead. We can use some encoding to force everything into ascii